overlord/hookstate/hookmgr.go

```diff
@@ -218,11 +220,17 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 		output, err := runHook(context, tomb)
 		if err != nil {
 			err = osutil.OutputErr(output, err)
-			if handlerErr := context.Handler().Error(err); handlerErr != nil {
-				return handlerErr
+			if hooksup.IgnoreFail {
+				task.State().Lock()
+				task.Logf("ignoring failure in hook %q: %v", hooksup.Hook, err)
```

Comment:
I think you can use Errof here, is a kind of error even if it doesn't make the task fail